Apis

BeckySharp opened the pull request:
@maxaalexeeva as I was trying to get this to work, stumbled across some things that I *think* I cleaned up, but please check the changes and let me know if I am breaking some functionality accidentally.  if it looks good, please merge

fyi @cl4yton

maxaalexeeva commented on text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala:
```diff
@@ -124,28 +124,31 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
-      if word.length <= 6
-      if (word.toLowerCase != word // mixed case or all UPPER
+    def passesFilters(v: Mention, isArg: Boolean): Boolean = {
+      if ((v matches "Variable") && isArg) return true
```
@BeckySharp, the condition in line 128 does not work (a test broke) because I also need this check when the Variable label is actually assigned, e.g., the word 'for' is found as a Variable by the rule 'var_calculation' because the word 'calculation' is frequently preceded by a variable (e.g., _Kcbmin calculation is implemented by..._), then the action looksLikeAVariable is run, and I know that it's not really a Variable. Does that make sense? I got rid of the condition.

BeckySharp replied:
ah, poop.
Ok, I'll modify the taxonomy and add an entry for Variables found with a Gazetteer, which i need to pass everytime...

maxaalexeeva replied:
Oh, I see what you were going for with that...

maxaalexeeva commented on text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala:
```diff
@@ -124,28 +124,31 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+      if (v.words.length != 1) return false
+      // Else, the variable candidate has length 1
+      val word = v.words.head
+      if (word.length > 6) return false
```
@BeckySharp, added another condition from the previous version of the action.

BeckySharp commented:
ok @maxaalexeeva please check it out again, I think I fixed the issue, though I added some technical debt... :/
That said, if you're ok with this, i can then let ppl know the api is working

maxaalexeeva commented:
@BeckySharp, tests pass. I don't have any objections. 
Will this kind of variable need to be incorporated into the rules?

BeckySharp commented:
Nope, bc it's a child of Variable, so all the rules that operate over
Variables would accept these too.
Thanks for checking it out for me!  I'll go ahead and merge

On Sat, Jan 18, 2020 at 2:46 PM Maria (Masha) Alexeeva <
notifications@github.com> wrote:

> @BeckySharp <https://github.com/BeckySharp>, tests pass. I don't have any
> objections.
> Will this kind of variable need to be incorporated into the rules?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ml4ai/automates/pull/102?email_source=notifications&email_token=ABJCPCI3R4SYDV2CYALV4FTQ6N2E3A5CNFSM4KIACEX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJKCUXI#issuecomment-575941213>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABJCPCJKVTAVGHU4TAETBRDQ6N2E3ANCNFSM4KIACEXQ>
> .
>

maxaalexeeva commented:
Ah! I see! Sounds good!
